frontend/src/metabase/core/components/CheckBox/types.ts

```diff
@@ -0,0 +1,47 @@
+import { ChangeEvent, FocusEvent, HTMLAttributes, ReactNode } from "react";
+
+export interface CheckBoxInputProps {
+  size: number;
+}
+
+export interface CheckBoxContainerProps {
+  disabled: boolean | undefined;
+}
+
+export interface CheckBoxIconProps {
+  checked: boolean;
+  uncheckedColor: string;
+}
+
+export interface CheckBoxIconContainerProps {
+  checked: boolean | undefined;
+  size: number;
+  checkedColor: string;
+  uncheckedColor: string;
+}
+
+export interface CheckBoxLabelProps {
+  labelEllipsis: boolean;
+}
+
+export interface CheckBoxProps
+  extends Omit<HTMLAttributes<HTMLElement>, "onChange" | "onFocus" | "onBlur"> {
+  label?: ReactNode;
+  labelEllipsis?: boolean;
+  checked?: boolean;
+  indeterminate?: boolean;
+  disabled?: boolean;
+  size?: number;
+  checkedColor?: string;
+  uncheckedColor?: string;
+  autoFocus?: boolean;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+}
+
+export interface CheckboxTooltipProps {
+  condition: boolean;
```

Comment:
nit: this prop name could be more descriptive

BTW, maybe `Ellipsified` component can be useful here?

Reply:
Not using `Ellipsified` because:
- It will add a `Tooltip` component for every rows even when not necessary
- It is using an observer when resizing but mobile layout is to be reviewed